src/Text/Parsing/StringParser/Parser.purs

```diff
@@ -0,0 +1,105 @@
+-- | This module defines the `Parser` type of string parsers, and its instances.
+
+module Text.Parsing.StringParser.Parser where
```

Comment:
Then again maybe we should just leave the current contents of Text.Parsing.StringParser.CodeUnits where they are, as moving them is probably also an unnecessary breaking change.

Reply:
Per your original instructions, `CodeUnits` was to be moved into `Text.Parsing.StringParser`. However, doing that caused a cyclical dependency between that module and `Text.Parsing.StringParser.Combinators`. So, I moved this code out into its own module and then re-export it in `Text.Parsing.StringParser`. 

I think leaving `Text.Parsing.StringParser.CodeUnits` as is makes sense to reduce breakage. However, why have a separate module for that now that `CodePoints` won't be its own module? If we're going to make breaking changes, now is the time, right?

Reply:
Yeah, good point, now is the time to make breaking changes. Since putting the current contents of Text.Parsing.StringParser.CodeUnits into Text.Parsing.StringParser directly won't work, how about putting them back where they were before the change which added the CodeUnits and CodePoints modules: Text.Parsing.StringParser.String?

Reply:
I think I should implement this in a separate PR after this one gets merged. It'll make the diff easier to see.